python/src/spark_rapids_ml/core.py

```diff
@@ -696,60 +696,74 @@ def fitMultipleModels() -> List["_CumlModel"]:
         else:
             return super().fitMultiple(dataset, paramMaps)
 
-    def _try_stage_level_scheduling(self, rdd: RDD) -> RDD:
-        ss = _get_spark_session()
-        sc = ss.sparkContext
+    def _skip_stage_level_scheduling(self, spark_version: str, conf: SparkConf) -> bool:
+        """Check if stage-level scheduling is not needed,
+        return true to skip stage-level scheduling"""
 
-        if ss.version < "3.4.0":
-            self.logger.warning(
-                "Stage level scheduling in spark-rapids-ml requires spark version 3.4.0+"
+        if spark_version < "3.4.0":
+            self.logger.info(
+                "Stage-level scheduling in spark-rapids-ml requires spark version 3.4.0+"
             )
-            return rdd
-        elif not _is_standalone_or_localcluster(sc):
-            # Only standalone or local-cluster supports stage-level scheduling with dynamic
-            # allocation disabled.
-            self.logger.warning(
-                "Stage level scheduling in spark-rapids-ml only works on spark standalone or "
-                "local cluster mode"
-            )
-            return rdd
+            return True
 
-        executor_cores = sc.getConf().get("spark.executor.cores")
-        executor_gpu_amount = sc.getConf().get("spark.executor.resource.gpu.amount")
+        if not _is_standalone_or_localcluster(conf):
+            self.logger.info(
+                "Stage-level scheduling in spark-rapids-ml requires spark standalone or "
+                "local-cluster mode"
+            )
+            return True
+
+        executor_cores = conf.get("spark.executor.cores")
+        executor_gpus = conf.get("spark.executor.resource.gpu.amount")
+        if executor_cores is None or executor_gpus is None:
+            self.logger.info(
+                "Stage-level scheduling in spark-rapids-ml requires spark.executor.cores, "
+                "spark.executor.resource.gpu.amount to be set."
+            )
+            return True
 
-        if executor_cores is None or executor_gpu_amount is None:
-            self.logger.warning(
-                "Stage level scheduling in spark-rapids-ml requires spark.executor.cores, "
-                "spark.executor.resource.gpu.amount to be set "
+        if int(executor_cores) == 1:
+            # there will be only 1 task running at any time.
+            self.logger.info(
+                "Stage-level scheduling in spark-rapids-ml requires spark.executor.cores > 1 "
             )
-            return rdd
+            return True
 
-        if int(executor_gpu_amount) > 1:
-            # For spark.executor.resource.gpu.amount>1, we suppose user knows how to configure
+        if int(executor_gpus) > 1:
```

Comment:
I don't think spark rapids supports executor_gpus > 1.   But shouldn't we allow this case for stage level scheduling without spark rapids for etl?

Reply:
For now, we can't do anything for this case because of the known spark bug.